MUICore/muicore/mui_server.py

```diff
@@ -354,13 +355,30 @@ def CheckManticoreRunning(
             )
         )
 
+    def StopServer(
+        self, request: StopServerRequest, context: _Context
+    ) -> StopServerResponse:
+        for mwrapper in self.manticore_instances.values():
+            mwrapper.manticore_object.kill()
+            stime = time.time()
+            while mwrapper.manticore_object.is_running():
+                time.sleep(1)
+                if (time.time() - stime) > 10:
+                    break
```

Comment:
Does this logic imply that Manticore was unable to be killed? I think a log message about what's happening at this point should be included.